gleam/src/typ.rs

```diff
@@ -557,6 +557,26 @@ pub enum ValueConstructorVariant {
 
     /// A function belonging to the module
     ModuleFn { module: Vec<String>, arity: usize },
+
+    /// A named struct
+    NamedStruct {},
+}
+
+impl ValueConstructorVariant {
+    fn to_module_value_constructor(&self) -> ModuleValueConstructor {
+        match self {
+            ValueConstructorVariant::Enum { .. } => ModuleValueConstructor::Enum,
+            ValueConstructorVariant::LocalVariable { .. } => ModuleValueConstructor::Fn,
+            ValueConstructorVariant::ModuleFn { .. } => ModuleValueConstructor::Fn,
+            ValueConstructorVariant::NamedStruct { .. } => unimplemented!(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum ModuleValueConstructor {
+    Enum,
+    Fn,
```

Comment:
We need a test for modules that export structs to show that they are rendered correctly when used in other modules (i.e. not as an enum or fn)